service/controller/app/v1/resource/kubeconfigfinalizer/create.go

```diff
@@ -53,18 +53,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	finalizerTag := key.KubeConfigFinalizer(cr)
 
 	if !contains(kubeConfig.Finalizers, finalizerTag) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting finalizer for kubeconfig %#q in namespace %#q", name, namespace))
-
-		kubeConfig.Finalizers = append(kubeConfig.Finalizers, finalizerTag)
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finalizer already clear for kubeconfig secret %#q in namespace %#q", name, namespace))
+	} else {
+		// TODO: Before we remove this resource, let's delete all finalizers that we put on kubeconfig secrets
```

Comment:
TODO is okay in here? just for a sign that I need to do it by today 😆

Reply:
If this is a short lived TODO can we just do it before we merge this?

Reply:
I didn't make it clear; What I would like to put in TODO list is removing this resource from app-operator.

Reply:
Sorry @tomahawk28 I'm still confused here. How will the finalizers be removed?

Also I'm fine with adding a TODO if you think its necessary. But please follow the process and create an issue. https://github.com/giantswarm/fmt/blob/master/go/todo_comments.md

Reply:
OK I think I get this now. So we merge this to remove the finalizers. After that we can do a follow up PR to remove the resource. Is that correct?